chat-plugins/blackjack.js

```diff
@@ -0,0 +1,501 @@
+			if ((points + 11) <= 21) {
+				points += 11;
+			} else {
+				points += 1;
+			}
+		}
+		return points;
+	}
+	next() {
+		if (this.timer) {
+			clearTimeout(this.timer);
+			this.timer = null;
+		}
+		if (Object.keys(this.players)[Object.keys(this.players).length - 1] === this.curUser && this.players[this.curUser].status !== 'playing') {
+			if (this.dealer.points < 17) {
+				this.giveCard('dealer');
+			} else if (this.dealer.points >= 17) {
+				let cards = '';
+				for (let u in this.dealer.cards) cards += `[${this.dealer.cards[u]}] `;
+				this.display(`<br /><strong>${this.dealer.name}</strong> stands with ${cards} (${this.dealer.points})`);
+				this.end();
+			}
+			return;
+		}
+		if (this.players[this.curUser].status !== 'playing') {
+			let number = 0;
+			while (this.players[Object.keys(this.players)[number]].status !== 'playing') number++;
+			this.curUser = Object.keys(this.players)[number];
+		}
+		let output = `|uhtml${this.players[this.curUser].selfUhtml}|user-blackjack-${this.blackjackNumber}|<div class="infobox">`;
+		output += `It's your turn to move, ${this.players[this.curUser].name}<br />`;
+		for (let u in this.players[this.curUser].cards) {
+			let card = this.players[this.curUser].cards[u];
+			output += `<img src="${CARD_IMAGE_PATH}${toId(card).toUpperCase() + this.symbols[card.substr(-1)]}.png" title="${card}" height="100"/>' `;
+		}
+		output += `<br />Score: ${this.players[this.curUser].points}${(this.players[this.curUser].points === 21 ? ` (you have blackjack!)` : ``)}`;
+		output += '<br /><button class="button" name="send" value="/blackjack hit" title="Hit (get another card)">Hit</button> | <button class="button" name="send" value="/blackjack stand" title="Stand (just keep these cards)">Stand</button>';
+
+		this.players[this.curUser].sendRoom(`|notify|Blackjack (${this.room.title})|It's your turn to play!`);
+		this.players[this.curUser].sendRoom(output);
+		this.timer = setTimeout(() => {
+			let cards = '';
+			for (let u in this.players[this.curUser].cards) cards += `[${this.players[this.curUser].cards[u]}] `;
+			this.players[this.curUser].status = 'stand';
+			this.display(Chat.html`<br /><strong>${this.players[this.curUser].name}</strong> stands with ${cards} (${this.players[this.curUser].points}) (Auto-stand: took too long to move)`);
+			this.next();
+		}, this.turnTimeoutMinutes * 60 * 1000);
+	}
+}
+
+class BlackjackPlayer extends Rooms.RoomGamePlayer {
+	constructor(user, game) {
+		super(user, game);
+
+		this.cards = [];
+		this.points = 0;
+		this.status = 'playing';
+		this.selfUhtml = '';
+	}
+}
+
+exports.commands = {
+	bj: 'blackjack',
+	blackjack: {
+		new: 'create',
+		create: function (target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (room.game) return this.errorReply("There is already a game running in this room.");
+			if (room.blackjackDisabled) return this.errorReply("Blackjack is currently disabled in this room.");
```

Comment:
Don't mutate rooms like this! Rooms are already megamorphic as they stand. Either add this to `ChatRoom`'s constructor or create a settings file for this